src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c

```diff
@@ -21,69 +63,1180 @@ const ActorInit Boss_Fd2_InitVars = {
+                Audio_SetBGM(0x100100FF);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_VALVAISA_DEAD);
+                func_80032C7C(globalCtx, &this->actor);
+            } else if (toucherFlags != 0) {
+                BossFd2_SetupDamaged(this, globalCtx);
+                this->damageFlashTimer = 10;
+                this->invulnerabilityTimer = 100;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_VALVAISA_DAMAGE1);
+            }
+            if (toucherFlags != 0) {
+                for (i = 0; i < 30; i++) {
+                    Vec3f sp64 = { 0.0f, 0.0f, 0.0f };
+                    Vec3f sp58 = { 0.0f, -1.0f, 0.0f };
+                    Vec3f sp4C;
+                    sp64.x = Math_Rand_CenteredFloat(6.0f);
+                    sp64.y = Math_Rand_ZeroFloat(4.0f) + 6.0f;
+                    sp64.z = Math_Rand_CenteredFloat(6.0f);
+                    sp4C.x = this->actor.posRot2.pos.x;
+                    sp4C.y = this->actor.posRot2.pos.y;
+                    sp4C.z = this->actor.posRot2.pos.z;
+                    BossFd2_SpawnSkullPiece(globalCtx, bossFd->particles, &sp4C, &sp64, &sp58,
+                                            (s16)Math_Rand_ZeroFloat(6.0f) + 10);
+                }
+            }
+        }
+    }
+}
+
+void BossFd2_UpdateFace(BossFd2* this, GlobalContext* globalCtx) {
+    f32 phi_f0;
+    f32 phi_f2;
+    s16 sp24[5] = { 0, 1, 2, 2, 1 };
+
+    if (((this->varianceTimer % 8) == 0) && (Math_Rand_ZeroOne() < 0.3f)) {
+        this->blinkTimer = 4;
+    }
+    if ((this->actionFunc == BossFd2_Vulnerable) || (this->actionFunc == BossFd2_Damaged)) {
+        if (this->varianceTimer & 0x10) {
+            this->eyeState = 1;
+        } else {
+            this->eyeState = 2;
+        }
+    } else {
+        this->eyeState = sp24[this->blinkTimer];
+    }
+    DECR(this->blinkTimer);
+
+    if (this->screamTimer != 0) {
+        phi_f0 = 6000.0f;
+        phi_f2 = 1300.0f;
+    } else {
+        phi_f0 = (this->varianceTimer & 0x10) ? 1000.0f : 0.0f;
+        phi_f2 = 700.0f;
+    }
+    Math_SmoothScaleMaxF(&this->jawOpening, phi_f0, 0.3f, phi_f2);
+
+    DECR(this->screamTimer);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D285C.s")
+void BossFd2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BossFd2* this = THIS;
+    s16 i;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D2930.s")
+    osSyncPrintf("FD2 move start \n");
+    this->disableAT = 0;
+    this->actor.flags &= ~0x400;
+    this->varianceTimer++;
+    this->unkTimer++;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D2A08.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/BossFd2_Init.s")
+    for (i = 0; i < 5; i++) {
+        if (this->timers[i] != 0) {
+            this->timers[i]--;
+        }
```

Comment:
does `DECR(this->timers[i])` work? (not `this->timers[i] = DECR(this->timers[i])`)

Reply:
for some reason BossFd2 doesn't like DECR. I'll give it another shot, but it's very strange.

Reply:
nope. while it can be matched, it's all sorts of fake. probably going to leave as is.